### PR TITLE
lab5 example

### PR DESCRIPTION
**Pull request by MinjiJeon**

### CODE REVIEW

File: lab5.R [hunk 1]
```diff
@@ -1,53 +1,46 @@
-## Homework 4: Functions to be debugged
+## Homework 4: Functions to be debugged MINJI
 ## Lab 5: Code review
 
 ## 1. Berhu penalty
 
 berhu <- function(x, delta) {
-  if(abs(x) < delta) {
+  if((abs(x) < delta)|(abs(x) == delta)) {
```
Comment (johneverettkoo) [inline lab5.R:7]: Put spaces around `|`

File: lab5.R [hunk 2]
```diff
@@ -1,53 +1,46 @@
...
   }
 }
 
-
 ## 2. Trimmed mean
 
 trimmed_mean <- function(x, trim) {
-  qlo = quantile(x, probs = (1 - trim) / 2)
-  qhi = quantile(x, probs = 1 - (1 - trim) / 2)
+  qlo = quantile(x, probs = trim/2)
```
Comment (johneverettkoo) [inline lab5.R:17]: Add spaces around `/` and other mathematical operators

File: lab5.R (hunk 1 above)
Comment (MinjiJeon) [reply lab5.R:7]: Yay

File: lab5.R [hunk 3]
```diff
@@ -62,28 +55,26 @@ process_data_line = function(str) {
...
 
-
 ## 4. Gradient descent
 gradient_descent <- function(fn, deriv, start, step_size, epsilon) {
   x = start
   while(TRUE) {
-    new_x = x + step_size * deriv(x)
-    if(abs(deriv(new_x)) <= epsilon) {
+    new_x = x - step_size * deriv(x) 
+    if(abs(fn(new_x) - fn(x)) < epsilon) {
```
Comment (johneverettkoo) [inline lab5.R:63]: This is one way to check to see if you should stop the loop, but the more common method is `if(abs(deriv(new_x)) <= epsilon)`

**Pull request merged**